Scripts/Flag_placer.py

```diff
@@ -0,0 +1,352 @@
+from statistics import mean, median
+import os
+import argparse
+
+parser = argparse.ArgumentParser()
+parser.add_argument('--bam', '-b', required=True, type=str, help='Path to bam file.')
+parser.add_argument('--output', '-o', required=False, type=str, help='Path to output folder.')
+parser.add_argument('--log', '-l', required=False, default=False, type=bool,
+                    help='Bool specifying if logfile should be made.')
+parser.add_argument('--threshold', '-t', required=False, default=0, type=int, help='int specifying the minimun amount'
+                                                                                    'of reads before a flag is made')
+parser.add_argument('--minpercentage', '-mp', required=False, default=0, type=float, help='float specifying the'
+                                                                                          'threshold for the minimum'
+                                                                                          'percentage of total reads in'
+                                                                                          'region before flagged.')
+parser.add_argument('--region', '-r', required=False, default='', type=str, help='String specifying the region in '
+                                                                                 'format: "chr#:start-stop". use '
+                                                                                 'chr#:0-0 for whole chromosome.')
+parser.add_argument('--high_insert_size', '-hi', required=False, default=500, type=int,
+                    help='Length of insert size to be classified as high.')
+parser.add_argument('--name', '-n', required=False, default='output', type=str, help='Name for the project. This is'
+                                                                                         'the name of the output file.')
+
+args = parser.parse_args()
+
+
+def fetch_reads():
+    """ The fetch_reads function fetches the reads from the bam file
+
+    :return reads: Pysam object containing read information
+    """
+    bamfile = pysam.AlignmentFile(args.bam, 'rb')
+
+    if args.region == 'all':
+        reads = bamfile.fetch()
+
+    else:
+        chromosome = args.region.split(':')[0].replace('chr', '')
+        start = args.region.split(':')[1].split('-')[0]
+        end = args.region.split(':')[1].split('-')[1]
+
+        if start == '0' and end == '0':
+            reads = bamfile.fetch(chromosome)
+
+        else:
+            reads = bamfile.fetch(chromosome, int(start), int(end))
+
+    return reads
+
+
+def place_flags(reads):
+    """ The place_flags function gets the coordinates of interesting regions and returns the coordinates of the flags
+    with their information.
+
+    :param reads: Pysam object containing read information
+    :return all_flags: A 2d list containing the coordinates of the flags and additional information.
+    """
+    all_flags = []
+    read_data = [0, 0, 0]
```

Comment:
Even commentaar toevoegen wat op welke positie komt te staan of een dict gebruiken?